awx/main/managers.py

```diff
@@ -160,7 +160,10 @@ def get_or_register(self):
             from awx.main.management.commands.register_queue import RegisterQueue
 
             pod_ip = os.environ.get('MY_POD_IP')
-            registered = self.register(ip_address=pod_ip)
+            if settings.IS_K8S:
+                registered = self.register(ip_address=pod_ip, node_type='control')
+            else:
+                registered = self.register(ip_address=pod_ip)
```

Comment:
@shanemcd this split may not be necessary if `AWX_AUTO_DEPROVISION_INSTANCES` is 100% synonymous with `IS_K8S`... and maybe it is, but I don't have a firm understanding of all of these knobs.

Reply:
I think it technically is at this point, but I can see a scenario in the future where we want to use `AWX_AUTO_DEPROVISION_INSTANCES` in VM-land.